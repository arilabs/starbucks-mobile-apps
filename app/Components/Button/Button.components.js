/**
 * Button Global
 * @param {object} styleButton
 * @param {function} onPress
 * @param {object} styleTitle
 * @param {string} sourceBtnImage
 * @param {object} styleBtnImage
 */

import React from 'react';
import {
  View,
  Text,
  Image,
  TouchableOpacity
} from 'react-native';
import PropTypes from 'prop-types';

const Button = (props) => (
  <TouchableOpacity
    style={props.styleButton}
    onPress={props.onPress}
  >
    <Text style={props.styleTitle}>{props.title}</Text>
    <Image
      source={props.sourceBtnImage}
      style={props.styleBtnImage}
    />
  </TouchableOpacity>
);

export default Button;

Button.propTypes = {};
