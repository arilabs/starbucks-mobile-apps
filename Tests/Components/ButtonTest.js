import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../app/Components/Button/Button.components';
import { Images } from '../../app/Themes/Images';

test('Button component renders correctly if title props no set', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button component renders correctly if title props is set', () => {
  const tree = renderer.create(<Button title="Ini Title" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button component renders correctly if title and image props set', () => {
  const tree = renderer.create(<Button title="Button Title" sourceBtnImage={Images.icSearch} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button component renders correctly if title, images, and onPress is set', () => {
  const tree = renderer.create(
    <Button
      title="Ini Button Title"
      sourceBtnImage={Images.icSearch}
      onPress={() => window.alert('Terpanggil')}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})