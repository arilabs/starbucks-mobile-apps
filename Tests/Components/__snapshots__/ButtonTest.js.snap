// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly if title and image props set 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Button Title
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../app/Assets/Images/ic_search.png",
      }
    }
  />
</View>
`;

exports[`Button component renders correctly if title props is set 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Ini Title
  </Text>
  <Image />
</View>
`;

exports[`Button component renders correctly if title props no set 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
  <Image />
</View>
`;

exports[`Button component renders correctly if title, images, and onPress is set 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Ini Button Title
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../app/Assets/Images/ic_search.png",
      }
    }
  />
</View>
`;
